stratum/hal/lib/common/hal_test.cc

```diff
@@ -64,6 +67,12 @@ class FakeProcmonService final : public procmon::ProcmonService::Service {
 };
 
 class HalTest : public ::testing::Test {
+ public:
+  void ShutdownExternalServerAndThread() {
+    CHECK_ERR(pthread_cancel(hal_->external_server_shutdown_thread_));
```

Comment:
`pthread_cancel` for thread management is an anti pattern. Also does not work with C++11 threads.